### PR TITLE
M5505: As BBMRI-ERIC I want facets for collection type and data type

### PR DESCRIPTION
**Pull request by fdlk**
Keeps the structure of the store in place but replaces the specific checkbox filter components with a single generic checkbox filter component.
Does not use the biobank plot filter components because they are not feature rich enough and bind too rigorously to the store.
Includes fixes for two small bugs that I ran into.

#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [ ] Code unit/integration/system tested
- [x] User documentation updated
- [x] Clean commits
- [x] No warnings during install
- [x] Updated flow typing
- [x] Added Feature/Fix to release notes

### CODE REVIEW

File: src/components/BiobankExplorerContainer.vue [hunk 1]
```diff
@@ -42,13 +42,16 @@
     },
     computed: {
       /* A computed 'compound' object for watching all filters */
+      // TODO: Use an action to retrieve the collections when the filters are changed
```
Comment (fdlk) [inline src/components/BiobankExplorerContainer.vue]: I think this is a separate issue, so I've kept things as is, but changing the filters should be an action that triggers this.

File: src/components/filters/ActiveFilterList.vue [hunk 2]
```diff
@@ -41,9 +41,12 @@
       removeFilter (filterType, selectedFilterId) {
         const filters = utils.removeFilterFromFilterArrayById(this.activeFilters[filterType], selectedFilterId)
         this.$store.commit(UPDATE_FILTER, {name: filterType, filters: filters})
+        const value = filters.length === 0 ? undefined : filters.join(',')
+        this.$router.push({query: {...this.$store.state.route.query, [filterType]: value}})
       },
       resetAllFilters () {
         this.$store.commit(RESET_FILTERS)
+        this.$router.push({query: {}})
```
Comment (fdlk) [inline src/components/filters/ActiveFilterList.vue]: The only reason that the route got changed if you changed the filters was that the filter component was watching the filter. I've switched to explicitly updating the route if when the filters get updated.

File: src/store/actions.js [hunk 3]
```diff
@@ -118,7 +138,7 @@ export default {
 
     const query = helpers.createRSQLQuery(state)
 
-    api.get(`${COLLECTION_API_PATH}?num=10000&attrs=biobank${query}`).then(response => {
+    api.get(`${COLLECTION_API_PATH}?num=10000&attrs=~id,biobank${query}`).then(response => {
```
Comment (fdlk) [inline src/store/actions.js]: caused postgres bug when you filter on refEntity with id column

File: test/unit/specs/store/mutations.spec.js [hunk 4]
```diff
@@ -11,10 +11,11 @@ describe('store', () => {
           }
         }
 
-        const countries = ['NL', 'BE', 'DE']
+        const countries = [{id: 'NL', name: 'Nederland'}, {id: 'BE', name: 'Belgie'}, {id: 'DE', name: 'Duitsland'}]
```
Comment (fdlk) [inline test/unit/specs/store/mutations.spec.js]: countries are objects, not strings, so I've updated the test

File: src/components/filters/CheckboxFilters.vue [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+                 type="checkbox"
+                 v-model="selection"
+                 :id="option.id"
+                 :value="option.id">
+          {{ option.label }}
+        </label>
+      </div>
+
+      <p class="text-right" @click.prevent="toggleSliceOptions" v-if="maxVisibleOptions">
+        <small v-if="sliceOptions"><a href=""><i v-if="options"><i class="fa fa-caret-down"></i> {{ options.length - maxVisibleOptions }} more</i></a></small>
```
Comment (fdlk) [inline src/components/filters/CheckboxFilters.vue]: fixes a bug where the component writes `-4 more`

**Comment by marikaris** [comment]
On a small screen (my macbook screen), the first two pixels of the filters are cut off
screen shot 2018-08-15 at 08 13 22

**Comment by marikaris** [comment]
Could you please write tests for the actions and getters of the two new filters?

**Comment by fdlk** [comment]
Cannot reproduce the styling issue, is it new in this PR?

**Comment by marikaris** [comment]
I just noticed, I had my screen zoomed out to 90%, with that setting, it is still an issue, but I don't know how relevant it is then.

**Comment by marikaris** [comment]
The tests for the two new getters are still missing:
screen shot 2018-08-16 at 14 45 51

**Pull request merged**